packages/@aws-cdk/aws-eks/lib/cluster-resource-provider.ts

```diff
@@ -40,7 +40,14 @@ export interface ClusterResourceProviderProps {
     *
     * If not defined, a default layer will be used.
     */
-  readonly onEventLayer?: lambda.ILayerVersion;
+  readonly proxyAgentLayer?: lambda.ILayerVersion;
```

Comment:
Why the name change...?

Reply:
The reason for the name change is because the proxy-agent layer is now being used in both `onEventHandler` and `isCompleteHandler` Lambdas. Because of this change i've deprecated the original `onEventLayer` and created a new prop `proxyAgentLayer` since we will now be passing this prop into more than just the `onEventHandler` Lambda.

The `onEventLayer` prop was introduced [a few weeks ago (sept 24)](https://github.com/aws/aws-cdk/pull/16657) so it should not impact many users (if any). The prop would only be used if the user wishes to bundle the layer themselves with a custom proxy agent. 

This prop follows the [same user customization we allow with the kubectl handler](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-eks.Cluster.html#kubectllayer). However if we'd rather not allow users to override that layer then we may want to deprecate the `onEventLayer` prop since it is no longer going to be used.

Reply:
I think we're over optimizing here. Can we just leave everything as is as far as layers go and simply add the `NodeProxyAgentLayer` to the lambda's that needed and thats it?

Reply:
Sure thing. I'll remove the new prop and leave the `onEventLayer` prop. I'll also reconnect the `onEventLayer` to the `onEventLambda`.